src/main/java/de/tum/in/www1/artemis/service/NotificationSettingsService.java

```diff
@@ -0,0 +1,105 @@
+     * @return a map with key of NotificationType and value Boolean indicating which types are (de)activated by the user's notification settings
+     */
+    private Map<NotificationType, Boolean> convertNotificationOptionsToNotificationTypesWithActivationStatus(Set<NotificationOption> notificationOptions) {
+        Map<NotificationType, Boolean> resultingMap = new HashMap<>();
+        for (NotificationOption option : notificationOptions) {
+            NotificationType[] tmpNotificationTypes = this.findCorrespondingNotificationTypesForNotificationOption(option);
+            for (NotificationType type : tmpNotificationTypes) {
+                resultingMap.put(type, option.isWebapp());
+            }
+        }
+        return resultingMap;
+    }
+
+    /**
+     * This is the place where the mapping between NotificationOptions and NotificationType happens on the server side
+     * Each NotificationOption can be based on multiple different NotificationTypes
+     * @param notificationOption which corresponding NotificationTypes should be found
+     * @return the corresponding NotificationType(s)
+     */
+    private NotificationType[] findCorrespondingNotificationTypesForNotificationOption(NotificationOption notificationOption) {
+        switch (notificationOption.getOptionSpecifier()) {
+            case "notification.exercise-notification.exercise-created-or-started": {
+                return new NotificationType[] { NotificationType.EXERCISE_CREATED };
+            }
+            case "notification.exercise-notification.exercise-open-for-practice": {
+                return new NotificationType[] { NotificationType.EXERCISE_PRACTICE };
+            }
+            case "notification.exercise-notification.new-post-exercises": {
+                return new NotificationType[] { NotificationType.NEW_POST_FOR_EXERCISE };
+            }
+            case "notification.exercise-notification.new-answer-post-exercises": {
+                return new NotificationType[] { NotificationType.NEW_ANSWER_POST_FOR_EXERCISE };
+            }
+            case "notification.lecture-notification.attachment-changes": {
+                return new NotificationType[] { NotificationType.ATTACHMENT_CHANGE };
+            }
+            case "notification.lecture-notification.new-post-for-lecture": {
+                return new NotificationType[] { NotificationType.NEW_POST_FOR_LECTURE };
+            }
+            case "notification.lecture-notification.new-answer-post-for-lecture": {
+                return new NotificationType[] { NotificationType.NEW_ANSWER_POST_FOR_LECTURE };
+            }
+            case "notification.instructor-exclusive-notification.course-and-exam-archiving-started": {
+                return new NotificationType[] { NotificationType.EXAM_ARCHIVE_STARTED, NotificationType.COURSE_ARCHIVE_STARTED };
+            }
+            default: {
+                return new NotificationType[0];
+            }
+        }
+    }
```

Comment:
I took a moment to look into this and it seems like this is causing a high complexity increase **+9** . 
I would suggest to maybe try out this [plugin](https://plugins.jetbrains.com/plugin/12159-codemetrics) to help you locally as I couldn't see the issue on [codacy](https://app.codacy.com/gh/ls1intum/Artemis/file/61236342259/issues/source?fileBranchId=25004975).

I looked into it a bit, and I think there's a much simpler way of doing this, with less code and complexity. Psuedocode below:

```
    Map<String, NotificationType[]> mappings = Map.of(
        "notification.exercise-notification.exercise-created-or-started", new NotificationType[]{EXERCISE_CREATED},
        "notification.exercise-notification.exercise-open-for-practice", new NotificationType[]{EXERCISE_PRACTICE},
        "notification.exercise-notification.new-post-exercises", new NotificationType[]{NEW_POST_FOR_EXERCISE},
        "notification.exercise-notification.new-answer-post-exercises", new NotificationType[]{NEW_ANSWER_POST_FOR_EXERCISE},
        "notification.lecture-notification.attachment-changes", new NotificationType[]{ATTACHMENT_CHANGE },
        "notification.lecture-notification.new-post-for-lecture", new NotificationType[]{NEW_POST_FOR_LECTURE }
    );

    private NotificationType[] findCorrespondingNotificationTypesForNotificationOption(NotificationOption notificationOption) {
        return mappings.getOrDefault(notificationOption.getOptionSpecifier(), new NotificationType[0]);
    }
```

So maybe have a predefined set of mappings, and then use map get. You can avoid,  NotificationType.NEW_ANSWER_POST_FOR_LECTURE and use NEW_ANSWER_POST_FOR_LECTURE  directly if you import it statically.

Reply:
You are 100% correct. I now refactored this and the corresponding part on the client-side by using maps.
Now the code is cleaner and complexity lower. Thank you 👍